status/types.go

```diff
@@ -106,9 +106,22 @@ func (x *Status) IterateDetails(f func(*Detail) bool) {
 	}
 }
 
-// SetDetails sets Detail list of the Status.
-func (x *Status) SetDetails(v []*Detail) {
+// ResetDetails empties the detail list.
+func (x *Status) ResetDetails() {
```

Comment:
do we need to have it public?

Reply:
Yes, it is used in `SetStatusDetails` function

Reply:
yes but that func is it that pkg so we can leave it private since I do not see the usage of unsettling details outside of the `sdk` repo

Reply:
I prefer to avoid using the private components in functions.